### ISSUE TITLE
Integrating Stripe payment with Sharetribe

### ISSUE THREAD

**Issue by usmanarif** [sharetribe/sharetribe#889]
Is the integration of Stripe payment Gateway is possible in existing Sharetribe open source application?

**Comment by chrigui** [comment]
you do not need marketplaces from stripe to use Stripe for Multi vendor platform you can use Stripe Connect, in fact it makes more sense to use Stripe connect over marketplaces because the site admin is not responsible for charge backs.
https://stripe.com/blog/stripe-connect
https://stripe.com/docs/connect#payments-flow

hope this helps.

**Comment by Kusti** [comment]
@chrigui you are correct, while Stripe's marketplace API (https://stripe.com/us/marketplaces) is not yet available to the users, the integration could be built with Stripe Connect. However, this would mean that each merchant would need to create their own Stripe account, which could be a bit annoying for sellers in some marketplaces (at least in eBay-style "selling used stuff" sites). Also, building the integration with Stripe Connect would mean that the marketplace cannot keep any money in escrow (like sites like Airbnb do), which could be an issue in some cases.

So, right now we're watching how things develop, and what would be the best way to offer support for a platform like Stripe. There are also other options besides Stripe that we're considering, like https://www.balancedpayments.com/ and https://www.mangopay.com/

**Comment by isaacsanders** [comment]
Would a contribution of Stripe Marketplace integration be accepted, permitting it follows code standards?

**Comment by chrigui** [comment]
@isaacsanders Stripe connect would be nice :P

**Comment by chrigui** [comment]
@Kusti : thats correct, it might be pain to try to have each vendor signup for stripe this also has the benefit of the marketplace is not going to be effected by bad feedback or disputes raised by buyers.

**Comment by Kusti** [comment]
@chrigui very true.

@isaacsanders thanks a lot for the offer, we really appreciate it! However, we're planning to introduce big changes to our payment architecture pretty soon, since we need to organize it a bit differently to make adding new gateways easier. I think that it might be a bit challenging to start working on tis right now, since things will be changing a lot in our end, which might mean that a lot of the work you would be doing would go to waste. 

We are trying to make it easier to add new payment integrations (that's one big goal of the change), but right now it's still a bit of a challenge. Also, regarding Stripe, I think we haven't fully made the decision on which one would be better for our platform Stripe Connect or Stripe Marketplace, and more research is needed on that work before we can truly estimate what is the right option for our platform.

**Comment by isaacsanders** [comment]
Excellent! I can't wait to see how the design matures!

**Comment by pablorodriguezgonzalez** [comment]
Hello!

Are there news regarding to this?

Stripe Connect are available in a lot of countries right now.

By the way, @kusty, I don't understand this " However, this would mean that each merchant would need to create their own Stripe account, which could be a bit annoying for sellers in some marketplaces" as with PayPal it's exactly the same, the merchant must create a PayPal business account, what I'm lossing?

Best.

**Comment by Kusti** [comment]
@pablorodriguezgonzalez no updates yet. Stripe Connect is available in quite a few countries, but actually in order to work with managed accounts (which is useful to many marketplaces, because then the sellers don't have to go to Stripe to create an account there), you need to be in US or Canada still.

You are correct, with PayPal the merchants also need to create a PayPal business account, which actually doesn't work very ideally for eBay-style online flea market sites, so we'd definitely like to improve that process. Stripe's managed accounts could be a way to do that.

You can follow this thread in our idea forum if you want to get updated when we start moving the Stripe integration forward: http://support.sharetribe.com/forums/240322-sharetribe-development-ideas/suggestions/6790422-integrate-with-stripe

**Comment by pablorodriguezgonzalez** [comment]
Hello @Kusti!

It isn't exact, while in the doc you are right and can see that is needed to be in US or Canada, Connect is available for other countries and with managed accounts, in Domesting we're in the UK and in Spain and working with managed :) If you need more information, I'm happy to help with that.

Best.

**Comment by Kusti** [comment]
@pablorodriguezgonzalez ok, cool! I just emailed Stripe support a few days ago and they told me that to use Connect with managed accounts the standalone account needs to be in US or Canada. Weird... anyway, I can't yet give a timeline for when we are able to move forward with Stripe integration, because there's lots of other stuff we need to figure out before that. But I'm sure we get there eventually. :)

**Comment by pablorodriguezgonzalez** [comment]
@Kusti isn't strange, sometimes happens depending on the person that answered to you :D 

So I recommend you to insist, if you need some help with that I can put you in touch with a person in Stripe, not support itself. And of course, if you need some technical help with managed accounts and others, happy to help too!

Best.

**Comment by pdgwien** [comment]
@pablorodriguezgonzalez Hello, we're trying to implement Stripe Connect in our marketplace too, but we're outside of US and Canada, so we cannot use those. Could you tell us, how you worked around that?

Thanks!

**Comment by JarleKristiansen** [comment]
@pdgwien 
I have just been in contact with stripe. And they confirm that company outside US can use Stripe Connect. Ouer payment flow is: money is put on hold/fund. 7 days after work have been done, money been paid out to provider - 10% that goes to the marketplace. 

Please send me a mail. Its better for all of us if we hier the same developer company, or share the cost of implement stripe payment from scratch. Anyone instred in that?
Share the cost of implement stripe to sharetribe open source?

**Comment by pablorodriguezgonzalez** [comment]
Hello,

We can do the implementation for a low cost budget, having a lot of
experience not only in Connect but also in new Stripe features like managed
accounts (better for Sharetribe) and others.

Best.

**Comment by JarleKristiansen** [comment]
Hi

 

Thanks for reaching out 

 

Are you a developer company?

 

Best regards

 

Jarle Kristiansen

Tlf. +47 93404021

CEO

www.GOstyling.com

 

Fra: pablorodriguezgonzalez [mailto:notifications@github.com] 
Sendt: 4. november 2015 14:34
Til: sharetribe/sharetribe
Kopi: JarleKristiansen
Emne: Re: [sharetribe] Integrating Stripe payment with Sharetribe (#889)

 

Hello,

We can do the implementation for a low cost budget, having a lot of
experience not only in Connect but also in new Stripe features like managed
accounts (better for Sharetribe) and others.

Best.

**Comment by JarleKristiansen** [comment]
In addition to the strip I would like geo localization

Please send me a mail if you can perform parts of tailor adjustment to the marketplace. I will give you more accurate details. Please contact me at jarle@viewinvest.no

**Comment by pablorodriguezgonzalez** [comment]
Hello again,

Yes, mainly developers of Domesting, that it's a booking system for
domestic services that includes Stripe since January. We use
PayPal/Braintree too.

Best.

**Comment by JarleKristiansen** [comment]
@pablorodriguezgonzalez 

In addition to the strip I would like geo localization

Please send me a mail if you can perform parts of tailor adjustment to the marketplace. I will give you more accurate details. Please contact me at jarle@viewinvest.no

**Comment by JarleKristiansen** [comment]
@pablorodriguezgonzalez 
ok, I'm interested in the stripe integration to the marketplace. 
I can not use Braintree, they does not alow to split payment. Can you give me an estimate of the cost of stripe integration to the marketplace?

**Comment by Adonaki** [comment]
any update on this?

**Comment by JarleKristiansen** [comment]
Thanks, yes it is in progress

**Comment by Adonaki** [comment]
Cool, any sort of timeline? :)

**Comment by ajnatural** [comment]
If anyone's interested, I do some freelance work and can get a Stripe + Stripe Connect Sharetribe integration done pretty quickly for them. Shoot me an email at anshul@menith.com.

**Comment by JarleKristiansen** [comment]
Hi

 

We are integrating.

But how long time you think it will take to set this up?

Some information. 

-          Marketplace take 10% cut, include all expencive

-          90% ispaid out to provider, 5 workingdays after job is done

-          If any complain customer and provider must try to find this out them self.

 

 

 

Best regards

 

Jarle Kristiansen

Tlf. +47 93404021

CEO

www.GOstyling.com

 

Fra: ajnatural [mailto:notifications@github.com] 
Sendt: 29. mars 2016 04:24
Til: sharetribe/sharetribe
Kopi: JarleKristiansen
Emne: Re: [sharetribe/sharetribe] Integrating Stripe payment with Sharetribe (#889)

 

If anyone's interested, I do some freelance work and can get a Stripe + Stripe Connect Sharetribe integration done pretty quickly for them. Shoot me an email at anshul@menith.com.

**Comment by ajnatural** [comment]
Can have it done in 2 working days. If you want to discuss further might be better to contact me directly through email so we don't litter this comment thread.

**Comment by pierogranja** [comment]
@pablorodriguezgonzalez 
please contact me
pierogranja|@hotmail.com

**Comment by delblues** [comment]
hi @ajnatural 
I need gateway integration on sharetribe, do you still available?
Gonna send you an e-mail (anshul@menith.com). Tks!

**Comment by yologo** [comment]
We also need stripe integration into sharetribe. Please email us at: goyologo@gmail.com

Basically we need multi currency support for split payments.

Any other options (Braintree ??) please let us know as well.

**Comment by hansgtbrd** [comment]
here is someone working on an option as well: http://mooreds.us4.list-manage.com/subscribe?u=ed3e1976af3b1e42efaf4d860&id=6146f089f1

If someone else is working on a system, pls let us know under hfgutbrod@gmail.com
